examples/typed/main.go

```diff
@@ -0,0 +1,60 @@
+package main
```

Comment:
The PR description mentions this can be a good use for multiple clusters. Could we add an example that expands on the current `reconcile.Request` to add Cluster information?

Reply:
I've added one under examples/multiclustersync